apiserver/facades/client/spaces/spaces.go

```diff
@@ -325,3 +410,41 @@ func (api *API) checkSupportsSpaces() error {
 	}
 	return nil
 }
+
+func (api *API) getMachineCountBySpaceID(spaceID string) (int, error) {
+	count := 0
+	machines, err := api.backing.AllMachines()
+	if err != nil {
+		return 0, err
+	}
+	for _, doc := range machines {
+		spaceAddMap, err := doc.AddressesBySpaceID()
+		if err != nil {
+			return 0, errors.Trace(err)
+		}
+		if _, found := spaceAddMap[spaceID]; found {
+			count++
+		}
+	}
+	return count, nil
+}
+
+func (api *API) getApplicationsBindSpace(spaceID string) ([]string, error) {
+	endpointBindings, err := api.backing.AllEndpointBindings()
+	if err != nil {
+		return nil, err
+	}
+	applicationsMap := make(map[string]bool)
```

Comment:
take a look at NewStrings() in "github.com/juju/collections/set", it does the functionality you want here for you.